## Description: When forwarding button presses to uinput, it is now necessary
##              to send a "SYN" message after each keypress.
## Origin/Author: Upstream LIRC project, commit d987ce9c68239f637dddb9de9f9668f6e8f9fc1d
## Bug: https://bugs.launchpad.net/ubuntu/+source/lirc/+bug/567512
--- a/daemons/lircd.c	2009-12-30 14:37:13.000000000 -0600
+++ b/daemons/lircd.c	2009-12-30 14:39:42.000000000 -0600
@@ -1909,6 +1909,17 @@
 				logprintf(LOG_ERR, "writing to uinput failed");
 				logperror(LOG_ERR, NULL);
 			}
+
+			/* Need to write sync event */
+			memset(&event, 0, sizeof(event));
+			event.type = EV_SYN;
+			event.code = SYN_REPORT;
+			event.value = 0;
+			if(write(uinputfd, &event, sizeof(event)) != sizeof(event))
+			{
+				logprintf(LOG_ERR, "writing EV_SYN to uinput failed");
+				logperror(LOG_ERR, NULL);
+			}
 		}
 	}
 #endif
